Nested required properties with dot in ConfigItem's name lead to compilation error

### Describe the bug

If we have two configuration classes
```java
@ConfigRoot(name = "my-config")
public class MyConfigs {
  /**
   * Named configurations
   */
  @ConfigItem(name = ConfigItem.PARENT)
  Map<String, MyConfig> namedConfigs;
}
```
and
```java
@ConfigGroup
public class MyConfig {
  /**
   * The all-mighty fooBar config
   */
  @ConfigItem(name = "foo.bar", defaultValue= "baz")
  String fooBar;
}
```

The compiler will complain when an application using the extension is compiled.

### Expected behavior

No compiler error

### Actual behavior

Compilation fails with

```bash
...
Caused by: java.util.NoSuchElementException: SRCFG00014: The config property quarkus.my-config."named".foo.foo.bar is required but it could not be found in any config source
	at io.smallrye.config.SmallRyeConfig.convertValue(SmallRyeConfig.java:294)
...
```

### How to Reproduce?

Steps to reproduce the behaviour:
1. checkout https://github.com/turing85/quarkus-extension-config-bug
2. run `./mvn install`

### Output of `uname -a` or `ver`

N/A

### Output of `java -version`

openjdk 17.0.4 2022-07-19

### GraalVM version (if different from Java)

N/A

### Quarkus version or git rev

2.13.1.FINAL

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)

### Additional information

This is caused by:
- the `"."` in `@ConfigItem(name = "foo.bar", defaultValue= "baz")` in `MyConfig.java` and
- the `@ConfigItem(name = ConfigItem.PARENT) Map<String, MyConfig> namedConfigs;` in `MyConfigs.java`.

If we replace all occurrences of `foo.bar` with `fooBar`, everything works as expected.

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -40,7 +40,7 @@
         <microprofile-jwt.version>1.2</microprofile-jwt.version>
         <microprofile-lra.version>1.0</microprofile-lra.version>
         <smallrye-common.version>1.13.2</smallrye-common.version>
-        <smallrye-config.version>2.12.2</smallrye-config.version>
+        <smallrye-config.version>2.13.0</smallrye-config.version>
         <smallrye-health.version>3.3.0</smallrye-health.version>
         <smallrye-metrics.version>3.0.5</smallrye-metrics.version>
         <smallrye-open-api.version>2.3.1</smallrye-open-api.version>

--- a/extensions/resteasy-classic/rest-client/deployment/src/test/java/io/quarkus/restclient/configuration/RestClientBuildTimeConfigSource.java
+++ b/extensions/resteasy-classic/rest-client/deployment/src/test/java/io/quarkus/restclient/configuration/RestClientBuildTimeConfigSource.java
@@ -17,7 +17,7 @@ import io.smallrye.config.common.MapBackedConfigSource;
  * runtime, but it doesn't return any configuration.
  */
 public class RestClientBuildTimeConfigSource extends MapBackedConfigSource {
-    // Because getPropertyNames() is called during SmallRyeConfig init (2 times)
+    // Because getPropertyNames() is called during SmallRyeConfig init (3 times)
     private int propertyNamesCallCount = 0;
 
     private static final Map<String, String> PROPERTIES = Map.of(
@@ -42,7 +42,7 @@ public class RestClientBuildTimeConfigSource extends MapBackedConfigSource {
 
     @Override
     public Set<String> getPropertyNames() {
-        if (propertyNamesCallCount >= 2) {
+        if (propertyNamesCallCount >= 3) {
             return isBuildTime() ? PROPERTIES.keySet() : emptySet();
         } else {
             propertyNamesCallCount++;

--- a/jakarta/rewrite.yml
+++ b/jakarta/rewrite.yml
@@ -567,7 +567,7 @@ recipeList:
       newValue: '3.0'
   - org.openrewrite.maven.ChangePropertyValue:
       key: smallrye-config.version
-      newValue: '3.0.0'
+      newValue: '3.1.0'
   - org.openrewrite.maven.ChangePropertyValue:
       key: smallrye-fault-tolerance.version
       newValue: '6.1.0'

